cwf/gateway/integ_tests/omni_rules_test.go

```diff
@@ -52,37 +55,62 @@ func TestOmnipresentRules(t *testing.T) {
+			MaxUsageTime:   GyMaxUsageTime,
+			ValidityTime:   GyValidityTime,
+			UseMockDriver:  true,
+		},
+	)
 	imsi := ues[0].GetImsi()
 
 	// Set a block all rule to be installed by the PCRF
 	err = ruleManager.AddStaticRuleToDB(getStaticDenyAll("static-block-all", "mkey1", 0, models.PolicyRuleConfigTrackingTypeONLYPCRF, 30))
 	assert.NoError(t, err)
-	// Override with an omni pass all static rule with a higher priority
-	err = ruleManager.AddStaticPassAllToDB("omni-pass-all-1", "", 0, models.PolicyRuleTrackingTypeNOTRACKING, 20)
+	// Override with an omni pass all static rule with a higher priority and with a charging key
+	err = ruleManager.AddStaticPassAllToDB("omni-pass-all-1", "", 1, models.PolicyRuleConfigTrackingTypeONLYOCS, 20)
 	assert.NoError(t, err)
 	// Apply a network wide rule that points to the static rule above
 	err = ruleManager.AddOmniPresentRulesToDB("omni", []string{"omni-pass-all-1"}, []string{""})
 	assert.NoError(t, err)
 	tr.WaitForPoliciesToSync()
 
-	usageMonitorInfo := getUsageInformation("mkey1", 1*MegaBytes)
-	initRequest := protos.NewGxCCRequest(imsi, protos.CCRequestType_INITIAL)
-	initAnswer := protos.NewGxCCAnswer(diam.Success).
+	// Gx - PCRF config
+	usageMonitorInfo := getUsageInformation("mkey1", 5*MegaBytes)
+	gxInitRequest := protos.NewGxCCRequest(imsi, protos.CCRequestType_INITIAL)
+	gxInitAnswer := protos.NewGxCCAnswer(diam.Success).
 		SetStaticRuleInstalls([]string{"static-block-all"}, []string{}).
 		SetUsageMonitorInfo(usageMonitorInfo)
-	initExpectation := protos.NewGxCreditControlExpectation().Expect(initRequest).Return(initAnswer)
-	expectations := []*protos.GxCreditControlExpectation{initExpectation}
-	assert.NoError(t, setPCRFExpectations(expectations, nil)) // we don't expect any update requests
+	gxInitExpectation := protos.NewGxCreditControlExpectation().Expect(gxInitRequest).Return(gxInitAnswer)
+	gxExpectations := []*protos.GxCreditControlExpectation{gxInitExpectation}
+	assert.NoError(t, setPCRFExpectations(gxExpectations, nil)) // we don't expect any update requests
+
+	// Gy - OCS config
+	quotaGrant := &fegprotos.QuotaGrant{
+		RatingGroup: 1,
+		GrantedServiceUnit: &fegprotos.Octets{
+			TotalOctets: 1 * MegaBytes,
+		},
+		IsFinalCredit: false,
+		ResultCode:    diam.Success,
+	}
+	gyInitRequest := protos.NewGyCCRequest(imsi, protos.CCRequestType_INITIAL)
+	gyInitAnswer:= protos.NewGyCCAnswer(diam.Success).SetQuotaGrant(quotaGrant)
+	gyInitExpectation := protos.NewGyCreditControlExpectation().Expect(gyInitRequest).Return(gyInitAnswer)
+	gyExpectations := []*protos.GyCreditControlExpectation{gyInitExpectation}
+	assert.NoError(t, setOCSExpectations(gyExpectations, nil))
 
-	tr.AuthenticateAndAssertSuccessWithRetries(imsi, 5)
+	tr.AuthenticateAndAssertSuccess(imsi)
 
-	req := &cwfprotos.GenTrafficRequest{Imsi: imsi, Volume: &wrappers.StringValue{Value: *swag.String("200k")}}
+	req := &cwfprotos.GenTrafficRequest{Imsi: imsi, Volume: &wrappers.StringValue{Value: *swag.String("200K")}}
```

Comment:
Not your change, but *swag.String("200K") is equivalent to "200K", I think

Reply:
oh that swag.String thing was there before. I didnt even questioned it wasn't needed. I was just playing with the values. I will change that